From 43593d894b09075efe1ce73e611ace7e50eb4df3 Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Wed, 8 Aug 2018 20:48:04 +0530
Subject: [PATCH 029/706] dts: add iommu-map property in the fsl-mc device node

This patch also adds the alias for the ENETC device

(cherry picked from commit 49b1e979ddbb9fafbc44eb8d72c317f93018ec2d)

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index d3898f6ee164..592df9c59457 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -12,6 +12,14 @@
 	#address-cells= <2>;
 	#size-cells= <2>;
 
+	aliases {
+		rcie_enetc = &rcie_enetc;
+	};
+
+	aliases {
+		rcie_enetc = &rcie_enetc;
+	};
+
 	cpus{
 		#address-cells= <0x1>;
 		#size-cells= <0x0>;
@@ -324,6 +332,21 @@
 				<0000 0 0 4 &gic 0 0 0 117 0x4>;
 	};
 
+       rcie_enetc: pcie@1F0000000{         /* rcie_enetc */
+               compatible= "pci-host-ecam-generic";
+               reg= <0x01 0xF0000000 0x0 0x100000>;
+               #address-cells= <3>;
+               #size-cells= <2>;
+               #interrupt-cells = <1>;
+               device_type= "pci";
+               bus-range= <0x0 0x0>;
+               iommu-map = <0 &smmu 23 3>;
+               ranges= <0x82000000 0x0 0x00000000  0x1 0xff000000  0x0 0x100000     /* non-prefetchable memory */
+                        0xc2000000 0x0 0x00000000  0x1 0xff100000  0x0 0x100000     /* prefetchable memory */
+                        0x82000000 0x0 0x00000000  0x1 0xff200000  0x0 0x100000     /* non-prefetchable memory */
+                        0xc2000000 0x0 0x00000000  0x1 0xff300000  0x0 0x100000>;   /* prefetchable memory */
+       };
+
 
 	soc {
 		compatible = "simple-bus";
@@ -380,6 +403,8 @@
 		};
 	};
 
+
+
 	i2c0: i2c@2000000 {
 		compatible = "fsl,vf610-i2c";
 		#address-cells = <1>;
-- 
2.17.1

